sky/engine/core/window/window.cc

```diff
@@ -67,17 +66,42 @@ void Window::UpdateWindowMetrics(const SkyDisplayMetrics& metrics) {
   });
 }
 
-void Window::DispatchEvent(Event* event) {
+void Window::DispatchEvent(const String& event_type, double time_stamp) {
   DartState* dart_state = library_.dart_state().get();
   if (!dart_state)
     return;
   DartState::Scope scope(dart_state);
 
   DartInvokeField(library_.value(), "_dispatchEvent", {
-    ToDart(event),
+    DartConverter<String>::ToDart(dart_state, event_type),
+    ToDart(time_stamp)
   });
 }
 
+void Window::DispatchPointerPacket(const pointer::PointerPacketPtr& packet) {
+  DartState* dart_state = library_.dart_state().get();
+  if (!dart_state)
+    return;
+  DartState::Scope scope(dart_state);
+
+  Dart_Handle data_handle = Dart_NewTypedData(Dart_TypedData_kByteData,
+                                             packet->GetSerializedSize());
+  if (Dart_IsError(data_handle))
+    return;
+
+  Dart_TypedData_Type type;
+  void* data;
+  intptr_t len;
+  if (Dart_IsError(Dart_TypedDataAcquireData(data_handle, &type, &data, &len)))
+    return;
+
+  packet->Serialize(data, len);
+
+  Dart_TypedDataReleaseData(data_handle);
+
+  DartInvokeField(library_.value(), "_dispatchPointerPacket", {data_handle});
```

Comment:
This `DartInvokeField` acts as an embedder entry point. `//sky/engine/bindings/dart_vm_entry_points.txt` needs to be updated.